### ISSUE TITLE
Plugin naming behaviour is partially unexpected.

### ISSUE THREAD

**Issue by Pfeil** [robertcorponoi/godot_rust_helper#3]
Hi, I've just tried out things a little and like it a lot! I have some feedback and also found at least one bug, I think. Note that I'm trying to make a plugin and therefore did not use the "new" command but the "plugin" command. Also, it is my first time fiddling with gdnative in general.

1. when I choose a plugin name like "godot_pdf_gui" (library folder name "godot_pdf_impl", it will generate a ".rs" file in the library source code, and in the lib.rs is a line `mod ;`, resulting in a compiler error. This is pretty sure a bug. If I did something wrong, an error should occur at generation, which did not happen.
2. when I choose a name like "PDF Reader" it will use the word "reader" for the module and generate a PDFReader struct. I feel that this is quiet a lot of logic that is being implied here. I think it should be communicated in the beginners guide or imply less logic in there (i.e. just name the module pdf_reader).
3. I think some information about what the folder names of the godot project and the library project will affect would be nice. For example, even though my plugin was named "PDF Reader" (in a folder named godot_pdf_plugin), the library is called libgodot_pdf_impl.so (the libraries folder name) in the end. Not a bad thing, but I think one would expect it to be named "libpdf_reader.so" or something. I was just a little surprised. Of course, thinking about `cargo new` it makes totally sense.
4. But the fact that my plugin folder is only named "plugins/reader" and not "plugins/pdf_reader" or something, is a little weird and I am not sure if this is intended.

OS: Ubuntu 20.04
Shell: fish

**Comment by robertcorponoi** [comment]
Interesting, definitely not intended behavior, I haven't really had must testing with the plugin creation so this is helpful and I'll take a look at it right away and get back to you.

**Comment by robertcorponoi** [comment]
@Pfeil Heya, sorry for the delay but I believe I've fixed all of the issues (aside from documentation) you've mentioned.

As a side note, and the reason this took so long, is because I've been working on a replacement for this tool so keep an eye out for that it should be out sometime this week. I never really thought this tool would be popular but now that is has picked up in popularity I've started to refactor it to make it actually production quality including the documentation.
